addons/account/models/account_bank_statement.py

```diff
@@ -217,6 +217,7 @@ def _get_statement_validity(self):
         previous = self.env['account.bank.statement'].search(
             [
                 ('first_line_index', '<', self.first_line_index),
+                ('first_line_index', '!=', False),
```

Comment:
Because domains != SQL. :)
`first_line_index` is a char and you may have a stored '' which is smaller than any string. `!= False` will exclude the empty string too.

Reply:
For domains: `x != False` means that x is not falsy. If you ask the ORM to write '', it will write ''. When reading it converts nulls to ''.
The ORM needs to generate a query equivalent to searching for all records and then filtering and we are not sure that the database does not contain '' even if the ORM would not write '', the database may still contain them (for example an SQL update written somewhere).

It is not ideal, but that is an issue for another time.

Reply:
Depending on the query generated, it could make the use of the index less efficient then.

Reply:
Not really, the inequality is more restrictive (and other conditions are added with an AND), so it will use the same index and filter further.

Reply:
Separated inside #190445